### PR TITLE
FIXED #89 blocking controller memory leak

### LINKED ISSUE #89: Memory leaks in BlockingBlurController
Hello, @Dimezis 

I'm using your librray in such way: Activity -> Fragment -> RecyclerView Item with blurred background.
So, when I replace fragment with recyclerview and blured items to another one I'm getting several leaks detected by LeakCanary.

[screenshot #1](https://user-images.githubusercontent.com/10329086/53895627-83384400-4043-11e9-8057-f1c7ff66cbf8.png)
[screenshot #2](https://user-images.githubusercontent.com/10329086/53895629-83384400-4043-11e9-8e09-1ccdd013c3dd.png)
[screenshot #3](https://user-images.githubusercontent.com/10329086/53895630-83d0da80-4043-11e9-9bb1-6ab38b5d5abf.jpg)

In my (and LeakCanary's) opinion the leak was caused by anonymous class in BlockingBlurController#deferBitmapCreation ([read here about it](https://janatechnology.wordpress.com/2016/12/30/dealing-with-memory-leaks-from-anonymous-classes-in-android/))

I tried to remove it at all and as result no leaks detected and lower memory consumption achieved. So I can suggest to make that class static.

```java
private static class SizeHandler implements ViewTreeObserver.OnGlobalLayoutListener {

        private WeakReference<View> blurView;
        private SizeHandlerListener listener;

        SizeHandler(View blurView, SizeHandlerListener listener) {
            this.blurView = new WeakReference<>(blurView);
            this.listener = listener;
        }

        @Override
        public void onGlobalLayout() {
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN) {
                blurView.get().getViewTreeObserver().removeOnGlobalLayoutListener(this);
            } else {
                legacyRemoveOnGlobalLayoutListener();
            }

            int measuredWidth = blurView.get().getMeasuredWidth();
            int measuredHeight = blurView.get().getMeasuredHeight();

            listener.onSizeClarified(measuredWidth, measuredHeight);
        }

        @SuppressWarnings("deprecation")
        void legacyRemoveOnGlobalLayoutListener() {
            blurView.get().getViewTreeObserver().removeGlobalOnLayoutListener(this);
        }
    }

    private interface SizeHandlerListener {
        void onSizeClarified(int measuredWidth, int measuredHeight);
    }
```

And call it from function like it:
```java
private void deferBitmapCreation() {
     new SizeHandler(blurView, this::init);
}
```

This solution fixed memory leak for me.

Regards,
Alexey Egin.

### PR DESCRIPTION
**Pull request by masc3d**

### CODE REVIEW

**Comment by masc3d** [comment]
#89

**Comment by Dimezis** [comment]
This will require user to `post` `setBlurEnabled` or `setBlurAutoUpdate` in case if they want to disable the blur as soon as the screen is shown (before the layout is finished). 
But overall it's still probably a better way than having an implicit posting inside of the lib.

File: library/src/main/java/eightbitlab/com/blurview/BlockingBlurController.java [hunk 1]
```diff
@@ -287,12 +287,7 @@ void setBlurEnabledInternal(boolean enabled) {
 
     @Override
     public BlurViewFacade setBlurEnabled(final boolean enabled) {
-        blurView.post(new Runnable() {
-            @Override
-            public void run() {
-                setBlurEnabledInternal(enabled);
-            }
-        });
+        setBlurEnabledInternal(enabled);
```
Comment (Dimezis) [inline library/src/main/java/eightbitlab/com/blurview/BlockingBlurController.java]: `internal*` methods can be removed and inlined, as they are now doing the same stuff

**Comment by masc3d** [comment]
> This will require user to `post` `setBlurEnabled` or `setBlurAutoUpdate` in case if they want to disable the blur as soon as the screen is shown (before the layout is finished).

why?

**Comment by Dimezis** [comment]
Without getting into details, it just uses a shared flag with other stuff during initialization.
TBH it's just kind of a defect I should have fixed long time ago, so nvm

**Pull request merged**